### ISSUE TITLE
dcm4che migration

### ISSUE THREAD

**Issue by Enet4** [bioinformatics-ua/dicoogle#486]
Dicoogle currently still uses dcm4che-2.x, which is deprecated. More than once plugin developers have opted in to embedding more recent versions of dcm4che for other tasks at the price of not being fully compatible with other core constructs. Sometimes they even wanted to use newer versions, but had trouble integrating them with Dicoogle, so they ended up making workarounds. 

Let this issue be used to track future updates to dcm4che and decide when and how to push these changes. We also need to make an assessment of how much of dcm4che is used, both in sdk and core. The core Dicoogle project may be able to contain multiple versions, but the API in the SDK is a critical integration point.

The possibilities that I can think of right now:

1. Migrate core and `sdk` to use dcm4che in Dicoogle 4; Simple and effective, leads to a rather bumpy plugin migration. 
2. Have both versions at core and extend `sdk` to support certain inputs and outputs in either version of dcm4che. This is basically requiring the SDK to have a common ground for all tasks related with DICOM manipulation, which is likely to require a lot of effort.
3. Split the SDK, have Dicoogle integrate with one or more SDKs. Might be just as complicated as (2) and increase maintenance burden. 

In any of these, there are multiple migration targets. We could start first by moving to dcm4che3 and then to dcm4che v5, or jump directly to the latest dcm4che.

**Comment by Enet4** [comment]
In my opinion, we should take the quick and painful approach of migrating everything to use the latest dcm4che at once, because the cost of migrating all core and plugins at once would pay off early and be less expensive than diluting that effort to subsequent versions.

**Comment by bastiao** [comment]
> In my opinion, we should take the quick and painful approach of migrating everything to use the latest dcm4che at once, because the cost of migrating all core and plugins at once would pay off early and be less expensive than diluting that effort to subsequent versions.

Yes, we have been defining and implementing a strong methodology for validation, this will give some safer guarantees to do it.

**Comment by phtran-dev** [comment]
Hi @Enet4 @bastiao ,
Just curiosity, what is the progress of migrating deprecated dcm4chee 2.x to new dcm4chee 5.x ? Can you please tell us the plan ?
